app/javascript/app/layouts/root/root-component.jsx

```diff
@@ -10,6 +10,35 @@ import Sticky from 'react-stickynode';
 import headerStyles from 'components/header/header-styles';
 import styles from './root-styles.scss';
 
+import ReactGA from 'react-ga';
+
+const { GOOGLE_ANALYTICS_ID } = process.env;
+
+function trackPage(page) {
+  ReactGA.set({ page });
+  ReactGA.pageview(page);
+}
+
+let gaInitialized = false;
+function handleTrack(location, prevLocation) {
+  if (GOOGLE_ANALYTICS_ID) {
+    if (!gaInitialized) {
+      ReactGA.initialize(GOOGLE_ANALYTICS_ID);
+      gaInitialized = true;
+    }
+    if (!prevLocation) {
+      trackPage(location.pathname);
+    } else {
```

Comment:
in both cases with end up with the same result --> `trackPage(location.pathname)`, I don't think we need that if here.
Could we also extract the initialize logic to a separate function to make it more readable? like that:
```
const initialize = () => {
  ReactGA.initialize(GOOGLE_ANALYTICS_ID);
  gaInitialized = true;
}
```
and then `if (!gaInitialized) { initialize() }`

Reply:
ah, but we're tracking page only if the page changed in the second case, right? so maybe we can simplify it to: 
```
const pageChanged = page !== prevPage;
if (!prevLocation || pageChanged) { trackPage(location.pathname) }
```